tools/fuzzer.cpp

```diff
@@ -0,0 +1,274 @@
+        builder.add(Value(ValueType::Null));
+        break;
+      case ADD_UINT64:
+        builder.add(Value(randomGenerator.mt64()));
+        break;
+      case ADD_INT64: {
+        uint64_t uintValue = randomGenerator.mt64();
+        int64_t intValue;
+        memcpy(&intValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(intValue));
+        break;
+      }
+      case ADD_DOUBLE: { //exception was thrown saying JSON doesn't support this type
+        uint64_t uintValue = randomGenerator.mt64();
+        double doubleValue;
+        memcpy(&doubleValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(doubleValue));
+        break;
+      }
+      case ADD_UTC_DATE:
+        builder.add(Value(randomGenerator.mt64(), ValueType::UTCDate));
+        break;
+      case ADD_BINARY:
+        builder.add(Value(std::bitset<64>(randomGenerator.mt64()).to_string()));
+        break;
+      case ADD_EXTERNAL: {
+        Builder extBuilder;
+        extBuilder.add(Value(randomGenerator.mt64()));
+        builder.add(Value(static_cast<void const *>(extBuilder.slice().start()), ValueType::External));
+        break;
+      }
+      case ADD_ILLEGAL:
+        builder.add(Value(ValueType::Illegal));
+        break;
+      case ADD_MIN_KEY:
+        builder.add(Value(ValueType::MinKey));
+        break;
+      case ADD_MAX_KEY:
+        builder.add(Value(ValueType::MaxKey));
+      default:
+        break;
+    }
+    break;
+  }
+}
+
+int main(int argc, char* argv[]) {
+  VELOCYPACK_GLOBAL_EXCEPTION_TRY
+    bool isTypeAssigned = false;
+    size_t numIterations = 1;
+    size_t numThreads = 1;
+    Format format = Format::VPACK;
+
+    int i = 1;
+    while (i < argc) {
+      char const *p = argv[i];
+      if (isOption(p, "--help")) {
+        usage(argv);
+        return EXIT_SUCCESS;
+      } else if (isOption(p, "--vpack") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::VPACK;
+      } else if (isOption(p, "--json") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::JSON;
+      } else {
+        int value;
+        std::from_chars(p, p + sizeof(p), value);
+        if (value >= 1) {
+          if (i == 3) { // as threads and iterations are optional arguments, threads must be preceeded by iterations
+            // in the command line, hence, argument 3
+            numThreads = value;
+          } else {
+            numIterations = value;
+          }
+        } else {
+          usage(argv);
+          return EXIT_FAILURE;
+        }
+      }
+      ++i;
+    }
+
+    size_t itsPerThread = numIterations / numThreads;
+    size_t leftoverIts = numIterations % numThreads;
+
+    auto threadCallback = [format](std::stop_token const& stoken, size_t iterations) {
+      try {
+        RandomGenerator randomGenerator;
+        while (iterations-- > 0) {
+          if (stoken.stop_requested()) {
+            return;
+          }
+          Builder builder;
```

Comment:
A large performance improvement will be possible by moving this Builder outside of the loop and reusing the instance in each iteration. You have to `clear()` the Builder in every iteration of course.